glocaltokens/client.py

```diff
@@ -405,14 +409,30 @@ def find_device(unique_id: str) -> NetworkDevice | None:
                         unique_id,
                     )
                     network_device = find_device(unique_id)
+                else:
+                    if item.device_name not in address_list:
+                        LOGGER.debug("%s not in address_list", item.device_name)
+                        continue
+
+                    ip_address = address_list.get(item.device_name) or "0.0.0.0"
+
+                    man_disc_device = NetworkDevice(
+                        name=item.device_name,
+                        ip_address=ip_address,
+                        port=0,
+                        model=item.hardware.model,
+                        unique_id=item.device_info.device_id,
+                    )
 
                 device = Device(
                     device_id=item.device_info.device_id,
                     device_name=network_device.name
                     if network_device is not None
                     else item.device_name,
                     local_auth_token=item.local_auth_token,
-                    network_device=network_device,
+                    network_device=network_device
+                    if network_device is not None
+                    else man_disc_device,
```

Comment:
We can have it a bit cleaner if we choose to go with `elif` as suggested above
```suggestion
                    network_device=network_device,
```